### ISSUE TITLE
Trigger onError hook while sending html error to user

### ISSUE THREAD

**Issue by nckcol** [fastify/help#206]
#### You have already researched for similiar issues?
yes

#### What are you trying to achieve or the steps to reproduce?
I want to log errors while sending html error to the user with error statusCode (>= 500).
Purpose of `onError` hook, described in docs, totally covers my case:
> This hook is useful if you need to do some custom error logging ...

But there is problem with "sends an error back to the user":
> This hook will be executed only after the customErrorHandler has been executed, and only if the customErrorHandler sends an error back to the user ...

So the question is how to create such response in `errorHandler` that is an Error (triggers onError hook for logging purposes) and responds with user friendly error page (but still with error status code) at the same time.
I can do such logging with `onResponse` hook for sure. But it will be better if i could use onError hook

```js
// global for all subapps error logging
fastify.addHook('onError', (request, reply, error, done) => {
  if(error.statusCode >= 500) {
    // i want to log all application errors here
    log.error(error.statusCode, error.message)
  }
  done();
})

// user accessible routes (html sub-app)
fastify.register((instance, options, done) => {
    // error handler for user-friendly error messages
    instance.setErrorHandler((error, request, reply, done) => {
      reply.code(500).type('text/html').send(fancyHtmlError);

      // also tried this:
      // error.message = fancyHtmlError;
      // reply.code(500).type('text/html').send(error)
      // but it serializes as JSON

      // it would be nice to have something like this:
      // reply.error(error).type('text/html').send(fancyHtmlError);
      done();
    });
    instance.get('/some-route', async () => {
      const error = new Error();
      error.message = "I want to log this message"
      error.statusCode = 500;
      throw error;
    });
    done();
});

// other sub-apps with their own errorHandlers
```

#### What was the result you received?
I can't use onError hook for logging if i want to respond with user-friendly errors

#### What did you expect?
I expect a way to define that response is an error for logging with `onError` hook, but still reply with html for users, instead of json.

#### Context

* *node version*: 12.13.0
* *fastify version*: 2.14.1
* *os*: Ubuntu 20.04 LTS 64-bit

Thank you!

**Comment by nckcol** [comment]
My current workaround is decorating `reply` with an `error` field in which I can put error object and log it with `onResponse` hook. But it seems a little bit ugly to me because i need to use both hooks, and cant go with just `onResponse` hook :slightly_frowning_face:
```
// global for all subapps error logging
fastify.decorateReply('error', null);
fastify.addHook('onError', (request, reply, error, done) => {
  if(error.statusCode >= 500) {
    log.error(error.statusCode, error.message)
  }
  done();
})
fastify.addHook('onResponse', (request, reply, done) => {
  if(reply.statusCode >= 500 && reply.error) {
    log.error(reply.error.statusCode, reply.error.message)
  }
  done();
})

// user accessible routes (html sub-app)
fastify.register((instance, options, done) => {
    // error handler for user-friendly error messages
    instance.setErrorHandler((error, request, reply, done) => {
      reply.code(500).type('text/html')
      reply.error = error;
      reply.send(fancyHtmlError);
      done();
    });
    
    // ...
});
```

**Comment by mcollina** [comment]
Thanks for letting us know! I don't know how to act on this immediately. It seems we might want to change when `'onError'` happens.

**Comment by mcollina** [comment]
I'm tranferring this to the main fastify repo.
